<a href="https://colab.research.google.com/github/PeaceUdoka/hamoyePythonProject/blob/main/hamoyeproject.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>


# HAMOYE PYTHON PROGRAMMING PROJECT TAG ALONG CODE



In [1]:
#importing libraries for data analysis
import pandas as pd

In [3]:
#loading data with latin-1 encoding to prevent UnicodeError generated otherwise
df = pd.read_csv('/content/FoodBalanceSheets_E_Africa_NOFLAG.csv', encoding = 'latin-1')

In [4]:
#showing first few rows of dataset
df.head()

Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   
4          4  Algeria       2901  Grand Total           684   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

### 11. What is the total sum of Animal Fat produced in 2014 and 2017 respectively?

In [17]:
df.groupby("Item")["Y2014"].sum()["Animal fats"]

209460.54

In [19]:
df.groupby("Item")["Y2017"].sum()["Animal fats"]

269617.53

 **Answer**  209460.54 and 269617.53

### 12. What is the mean and standard deviation across the whole dataset for the year 2015 to 3 decimal places?

In [20]:
df["Y2015"].mean()

135.23596649549626

In [21]:
df["Y2015"].std()

1603.4039839425013

**Answer** 
135.236 and 1603.404

### 13. What is the total number and percentage of missing data in 2016 to 2 decimal places?

In [23]:
df["Y2016"].isna().sum()

1535

In [25]:
df["Y2016"].isna().sum()/df["Y2016"].shape[0]*100

2.5187470259094566

**Answer** 1535 and 2.52

### 14. Which year had the highest correlation with ‘Element Code’?

In [33]:
df.corr()["Element Code"].sort_values()  #generates correlation for each column and sorts in ascending order to determine which year has the highest correation wigh "Element Code"

Item Code      -0.024683
Area Code      -0.000209
Y2016           0.023444
Y2015           0.023889
Y2017           0.024254
Y2018           0.024279
Y2014           0.024457
Element Code    1.000000
Name: Element Code, dtype: float64

**Answer** 2014

### 15. What year has the highest sum of Import Quantity? 

In [42]:
df.groupby("Element").sum().iloc[7,4:]

Y2015    267018.46
Y2016    286582.78
Y2017    294559.09
Y2018    287997.09
Name: Import Quantity, dtype: float64

In [43]:
df.groupby("Element").sum().iloc[7,4:].max()

294559.09

**Answer** 2017

### 16. What is the total number of the sum of Production in 2014?

In [45]:
df.groupby("Element")["Y2014"].sum()["Production"]

1931287.75

**Answer** 1931287.75

### 17. Which of these elements had the highest sum in 2018?

In [47]:
df.groupby("Element")["Y2018"].sum().sort_values()  #sorts the sum of all elements in 2018 in ascending order

Element
Tourist consumption                            90.00
Fat supply quantity (g/capita/day)          10258.69
Protein supply quantity (g/capita/day)      11833.56
Stock Variation                             20577.91
Seed                                        25263.14
Residuals                                   34864.00
Food supply quantity (kg/capita/yr)         49056.85
Other uses (non-food)                       91300.97
Losses                                     163902.00
Export Quantity                            181594.80
Feed                                       233489.68
Import Quantity                            287997.09
Processing                                 308429.00
Food supply (kcal/capita/day)              455261.00
Total Population - Both sexes             1140605.00
Food                                      1303841.28
Production                                2075072.89
Domestic supply quantity                  2161192.10
Name: Y2018, dtype: float64

**Answer** Domestic supply quantity

### 18. Which of these elements had the 3rd lowest sum in 2018?

**Answer** Protein supply quantity (g/capita/day)  *Check code fromm previous question*

### 19. What is the total Import Quantity in Algeria in 2018?

In [52]:
df.groupby(["Area", "Element"])["Y2018"].sum()["Algeria","Import Quantity"]

36238.29

**Answer** 36238.29

### 20. What is the total number of unique countries in the dataset?

In [53]:
df["Area"].nunique()

49

**Answer** 49